lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
In puppet language the distinction between how something is read is done by calling `file >> String` (assumed but not enforced UTF-8), or `binary_file >> Binary` (raw, bytes). We have however no control whatsoever over functions in the wild, and there is no specification other than the existing behavior where puppet is using the encoding embedded in the Ruby string. Note that the `file` read does not enforce UTF-8 - as it performs a binary read and any later requirement that the format must be UTF-8 makes things blow up later.